ark/utils/load_utils.py

```diff
@@ -249,85 +177,133 @@ def load_imgs_from_tree(data_dir, img_sub_folder=None, fovs=None, channels=None,
+        files (list):
+            list of files (e.g. ['fov1.tif'. 'fov2.tif'] to load.
+            If None, all (.tif, .jpg, .png) files in data_dir are loaded.
         delimiter (str):
-            character used to determine the file-prefix containging the fov name. Default is None.
+            character used to determine the file-prefix containging the fov name.
+            Default is None.
+        xr_dim_name (str):
+            sets the name of the last dimension of the output xarray.
+            Default: 'compartments'
+        xr_channel_names (list):
+            sets the name of the coordinates in the last dimension of the output xarray.
         dtype (str/type):
             data type to load/store
-        variable_sizes (bool):
-            Dynamically determine image sizes and pad smaller imgs w/ zeros
         force_ints (bool):
             If dtype is an integer, forcefully convert float imgs to ints. Default is False.
+        channel_indices (list):
+            optional list of indices specifying which channels to load (by their indices).
+            if None or empty, the function loads all channels.
+            (Ignored if data is not multitiff).
 
     Returns:
         xarray.DataArray:
-            xarray with shape [fovs, x_dim, y_dim, 1]
+            xarray with shape [fovs, x_dim, y_dim, tifs]
 
+    Raises:
+            ValueError:
+                Raised in the following cases:
+                 * data_dir is not a directory, <data_dir>/img is
+                not a file for some img in the input 'files' list, or no images are found.
+                * channels_indices are invalid according to the shape of the images.
+                * the provided dtype is too small to represent the data.
+                * The length of xr_channel_names (if provided) does not match the number
+                 of channels in the input.
     """
+    if not os.path.isdir(data_dir):
+        raise ValueError(f"Invalid value for data_dir. {data_dir} is not a directory.")
 
-    imgs = iou.list_files(data_dir, substrs=['.tif', '.jpg', '.png'])
-
-    imgs.sort()
+    if files is None:
+        imgs = iou.list_files(data_dir, substrs=['.tif', '.jpg', '.png'])
+    else:
+        imgs = files
+        for img in imgs:
+            if not os.path.isfile(os.path.join(data_dir, img)):
+                raise ValueError(f"Invalid value for {img}. "
+                                 f"{os.path.join(data_dir, img)} is not a file.")
 
     if len(imgs) == 0:
         raise ValueError(f"No images found in directory, {data_dir}")
 
     test_img = io.imread(os.path.join(data_dir, imgs[0]))
 
+    # check data format
+    multitiff = test_img.ndim == 3
+    channels_first = multitiff and test_img.shape[0] == min(test_img.shape)
+
+    # check to make sure all channel indices are valid given the shape of the image
+    # and that channels_names has the same length as the number of channels in the image
```

Comment:
I think we can move this second comment down to line 253, so that it comes right before the code it's describing

Reply:
done